TypeCobol.LanguageServer.Test/LSRTest.cs

```diff
@@ -388,6 +388,11 @@ public void GoToDefinition()
         }
 
         [TestMethod]
+#if DOMAIN_CHECKER
+        //This test does not pass CheckVariable with DOMAIN_CHECKER set because new semantic domain
```

Comment:
After investigation, it seems this test will duplicate the declaration of user :
See the working storage node:
![image](https://user-images.githubusercontent.com/13764482/70924538-fab2ef00-2029-11ea-9133-b5d6cf1cd2d3.png)
So the current behavior of this test is ok.

You can reproduce the bug with the following code:
```cobol
       IDENTIFICATION DIVISION.    
       PROGRAM-ID. DVZZBCO1.
       DATA DIVISION.
       WORKING-STORAGE SECTION.

       01 Person TYPEDEF STRICT PUBLIC.
           05 longName.
                10 firstName PIC X(20).

       01 User TYPEDEF STRICT PUBLIC.
           05 person    TYPE Person.


       01 User TYPEDEF STRICT PUBLIC.
           05 person    TYPE Person.

       01 user1         TYPE User.


       PROCEDURE DIVISION.
      *KO, variable user1::person::longName::firstName must no be 
	  *resolved because declaration of User is duplicate and "user1" 
	  *will not match a type
           MOVE "John" TO user1::person::longName::firstName
           .

       END PROGRAM DVZZBCO1.
```

Reply:
After further testing the exact code executed at the time of the difference between the current Variable Checking and the Seamantic Domain Variable Checking is the following:

```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. DVZZBCO1.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
      %<<<
           @desc: Contient les informations d'une personne
           @todo: Ajouter une variable Adresse
      %>>>
       01 Person TYPEDEF STRICT PUBLIC.
           05 longName.
                10 firstName PIC X(20).
                10 lastName  PIC X(20).
           05 birthDate PIC X(10).


       01 User TYPEDEF STRICT PUBLIC.
           05 userId    PIC X(20).
           05 passwd    PIC X(20).
           05 person    TYPE Person.

       01 User TYPEDEF STRICT PUBLIC.
           05 userId    PIC X(20).
           05 passwd    PIC X(20).
           05 person    TYPE Person.

       01 user1         TYPE User.
       01 tw            PIC 9.
       01 rc            PIC 9.

       01 uselessGroup.
           05 uselessVar1 PIC X.
           05 uselessVar2 PIC X.
           05 uselessVar3 PIC X.


       PROCEDURE DIVISION.
       INIT-LIBRARY.
           MOVE "John" TO user1::person::longName::firstName
           MOVE "DOE" TO user1::person::longName::lastName
      %<<
           Cette donnée ne nous sert à rien
           MOVE "01/01/1970" TO user1::person::birthDate
      %>>
           CALL WriteName              INPUT user1
                                       IN-OUT tw
                                       OUTPUT rc
                                       .
           CONTINUE.

       SOME-PARAGRAPH.
           MOVE "John" TO user1::person::longName::firstName
           .

      %<<<
           @params:
                -user: Toutes les informations de l'utilisateur
                -tw: La variable contenant les données pour écrire dans
                    un fichier
                -rc: Code retour de la fonction
           @desc: Ecrit sur une ligne le nom et
                  le prénom d'une personne
      %>>>
       DECLARE PROCEDURE WriteName
                INPUT user  TYPE DVZZBCO1::User.
                IN-OUT tw   PIC 9.
                OUTPUT rc   PIC 9.
       PROCEDURE DIVISION.
           CONTINUE.
       END-DECLARE.

       END PROGRAM DVZZBCO1.
```

Reply:
Thus you are right there is duplicated TypeDef instruction for the type User. The SemanticDomain also detect this situation in the ProgramSymbolTableBuilder, see screenshot below.

![image](https://user-images.githubusercontent.com/19149269/74842288-6ebf8b00-532a-11ea-8fb8-e8f74b5d2cde.png)

I am going write as you requested a specific test for that situation. The difference with the Semantic Domain is that in presence of a duplicated TypeDef it retains the first definition